script/model/training_util/data_transforming_util.py

```diff
@@ -44,63 +45,98 @@ def _tuple_num_memory_cardinality_linear_predict_transform(x, y):
+    new_y = y * (np.log2(tuple_num) + _TRANSFORM_EPSILON)[:, np.newaxis]
     # Transform linearly again based on the cardinality
     cardinality = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.EST_CARDINALITIES]])
     return new_y * cardinality[:, np.newaxis]
 
 
 # Transform the target linearly according to the tuple num
-_tuple_num_log_cardinality_linear_transformer = (_tuple_num_log_cardinality_linear_train_transform,
-                                                 _tuple_num_log_cardinality_linear_predict_transform)
+_num_log_cardinality_linear_transformer = (_num_log_cardinality_linear_train_transform,
+                                           _num_log_cardinality_linear_predict_transform)
 
 
-def _tuple_num_linear_log_train_transform(x, y):
+def _num_linear_log_train_transform(x, y):
     # Transform down the target according to the linear-log (nlogn) tuple num value in the input
     tuple_num = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.NUM_ROWS]])
-    return y / (tuple_num * np.log2(tuple_num) + 1)[:, np.newaxis]
+    return y / (tuple_num * np.log2(tuple_num) + _TRANSFORM_EPSILON)[:, np.newaxis]
 
 
-def _tuple_num_linear_log_predict_transform(x, y):
+def _num_linear_log_predict_transform(x, y):
     # Transform up the target according to the linear-log (nlogn) tuple num value in the input
     tuple_num = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.NUM_ROWS]])
-    return y * (tuple_num * np.log2(tuple_num) + 1)[:, np.newaxis]
+    return y * (tuple_num * np.log2(tuple_num) + _TRANSFORM_EPSILON)[:, np.newaxis]
 
 
 # Transform the target in a linear-log way (nlogn) according to the tuple num
-_tuple_num_linear_log_transformer = (_tuple_num_linear_log_train_transform, _tuple_num_linear_log_predict_transform)
+_num_linear_log_transformer = (_num_linear_log_train_transform, _num_linear_log_predict_transform)
 
 
-def _tuple_num_log_train_transform(x, y):
+def _num_log_train_transform(x, y):
     # Transform down the target according to the log tuple num value in the input
     tuple_num = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.NUM_ROWS]])
-    return y / (np.log2(tuple_num) + 1)[:, np.newaxis]
+    return y / (np.log2(tuple_num) + _TRANSFORM_EPSILON)[:, np.newaxis]
 
 
-def _tuple_num_log_predict_transform(x, y):
+def _num_log_predict_transform(x, y):
     # Transform up the target according to the log tuple num value in the input
     tuple_num = np.copy(x[:, data_info.INPUT_CSV_INDEX[ExecutionFeature.NUM_ROWS]])
-    return y * (np.log2(tuple_num) + 1)[:, np.newaxis]
+    return y * (np.log2(tuple_num) + _TRANSFORM_EPSILON)[:, np.newaxis]
 
 
 # Transform the target in a log scale (logn) according to the tuple num
-_tuple_num_log_transformer = (_tuple_num_log_train_transform, _tuple_num_log_predict_transform)
+_num_log_transformer = (_num_log_train_transform, _num_log_predict_transform)
+
+
+def _num_cardinality_linear_log_train_transform(x, y):
```

Comment:
Also, move "log" right after "tuple_num"/"num_rows", which directly refers to how the transformation is done. I.e., this should be `num_rows_log_cardinality_linear_train_transform`.